### PR TITLE
Android native share

### PR DESCRIPTION
**Pull request by ryanlntn**
Updates native-share with initial Android prototype flow.

- Adds support for text sharing
- Refactors to reuse NewGroupPage as initial route in share flow
- Starts refactor of navigation to start moving towards more sharing of navigation

Trello:
https://trello.com/c/mwQHXMCU/152-start-planning-android-changes-for-share
https://trello.com/c/a0MCWIiw/155-determine-and-implement-text-sharing-behavior

### CODE REVIEW

File: src/pages/NewChatPage.js [hunk 1]
```diff
@@ -186,7 +193,14 @@ function NewChatPage(props) {
         >
             {({didScreenTransitionEnd, safeAreaPaddingBottomStyle}) => (
                 <>
-                    <HeaderWithBackButton title={props.isGroupChat ? props.translate('sidebarScreen.newGroup') : props.translate('sidebarScreen.newChat')} />
+                    <HeaderWithBackButton
+                        title={
+                            share ? props.translate('newChatPage.shareToExpensify') : props.isGroupChat ? props.translate('sidebarScreen.newGroup') : props.translate('sidebarScreen.newChat')
+                        }
+                        shouldShowBackButton={!share}
```
Comment (cdanwards) [inline src/pages/NewChatPage.js:200]: LOL this is kinda just funny

**Review by cdanwards — APPROVED** [review]
Nice! As far as their components, looks like we're reusing pretty well. Lots of import shuffling

File: ios/share/Info.plist [hunk 2]
```diff
@@ -2,51 +2,52 @@
...
-        <key>NSExceptionDomains</key>
-        <dict>
-            <key>localhost</key>
-            <dict>
-                <key>NSExceptionAllowsInsecureHTTPLoads</key>
-                <true/>
-            </dict>
-        </dict>
-    </dict>
+		<dict>
```
[the remark this answers is not in the document]
Comment (ryanlntn) [reply ios/share/Info.plist:26]: I believe I updated these in XCode. That explains the tab indents.

File: src/Expensify.js [hunk 3]
```diff
@@ -168,11 +169,22 @@ function Expensify(props) {
         // Open chat report from a deep link (only mobile native)
         Linking.addEventListener('url', (state) => Report.openReportFromDeepLink(state.url, isAuthenticated));
 
+        const handleShare = (share) => {
```
[the remark this answers is not in the document]
Comment (ryanlntn) [reply src/Expensify.js:172]: Of course they do.

File: src/libs/Navigation/AppNavigator/AuthScreens.js [hunk 4]
```diff
@@ -1,38 +1,38 @@
+import lodashGet from 'lodash/get';
```
[the remark this answers is not in the document]
Comment (ryanlntn) [reply src/libs/Navigation/AppNavigator/AuthScreens.js:1]: Yeah. I originally had the new routes mounted here before I discovered that they need to be in a modal so `Navigate.dismissModal` will work.

**Pull request merged**